--- old/src/share/vm/runtime/globals.hpp	2015-10-14 14:23:27.000000000 +0300
+++ new/src/share/vm/runtime/globals.hpp	2015-10-14 14:23:27.000000000 +0300
@@ -838,6 +838,9 @@
   diagnostic(ccstrlist, DisableIntrinsic, "",                               \
          "do not expand intrinsics whose (internal) names appear here")     \
                                                                             \
+  product(bool, UseSpinLoopHintIntrinsic, false,                            \
+          "Use intrinsics for spinLoopHint")                                \
+                                                                            \
   develop(bool, TraceCallFixup, false,                                      \
           "Trace all call fixups")                                          \
                                                                             \
