--- old/src/share/vm/runtime/vmStructs.cpp	2015-10-14 14:23:28.000000000 +0300
+++ new/src/share/vm/runtime/vmStructs.cpp	2015-10-14 14:23:27.000000000 +0300
@@ -1857,6 +1857,7 @@
   declare_c2_type(StoreFenceNode, MemBarNode)                             \
   declare_c2_type(MemBarVolatileNode, MemBarNode)                         \
   declare_c2_type(MemBarCPUOrderNode, MemBarNode)                         \
+  declare_c2_type(SpinLoopHintNode, MemBarNode)                           \
   declare_c2_type(InitializeNode, MemBarNode)                             \
   declare_c2_type(ThreadLocalNode, Node)                                  \
   declare_c2_type(Opaque1Node, Node)                                      \
