--- old/src/share/vm/classfile/vmSymbols.hpp	2015-10-14 14:23:24.000000000 +0300
+++ new/src/share/vm/classfile/vmSymbols.hpp	2015-10-14 14:23:24.000000000 +0300
@@ -787,6 +787,9 @@
   do_intrinsic(_currentThread,            java_lang_Thread,       currentThread_name, currentThread_signature,   F_S)   \
    do_name(     currentThread_name,                              "currentThread")                                       \
    do_signature(currentThread_signature,                         "()Ljava/lang/Thread;")                                \
+  do_intrinsic(_spinLoopHint,             java_lang_Thread,       spinLoopHint_name, spinLoopHint_signature,     F_S)   \
+   do_name(     spinLoopHint_name,                               "spinLoopHint")                                        \
+   do_alias(    spinLoopHint_signature,                           void_method_signature)                                \
                                                                                                                         \
   /* reflective intrinsics, for java/lang/Class, etc. */                                                                \
   do_intrinsic(_isAssignableFrom,         java_lang_Class,        isAssignableFrom_name, class_boolean_signature, F_RN) \
