--- old/make/common/CORE_PKGS.gmk	2015-07-24 12:27:33.000000000 +0300
+++ new/make/common/CORE_PKGS.gmk	2015-07-24 12:27:33.000000000 +0300
@@ -39,7 +39,8 @@
     org.w3c.dom.css \
     org.w3c.dom.html \
     org.w3c.dom.stylesheets \
-    org.omg.stub.javax.management.remote.rmi
+    org.omg.stub.javax.management.remote.rmi \
+    jdk.performancehints
 
 #
 # ACTIVE_JSR_PKGS are packages that are part of an active JSR process--
@@ -297,4 +298,6 @@
     org.w3c.dom.views \
     org.xml.sax \
     org.xml.sax.ext \
-    org.xml.sax.helpers
+    org.xml.sax.helpers \
+    org.performancehints  
+
