--- old/src/cpu/sparc/vm/vm_version_sparc.cpp	2015-07-20 21:11:13.657992600 +0300
+++ new/src/cpu/sparc/vm/vm_version_sparc.cpp	2015-07-20 21:11:13.344301200 +0300
@@ -381,6 +381,12 @@
     FLAG_SET_DEFAULT(UseCRC32CIntrinsics, false);
   }
 
+  if (UseSpinLoopHintIntrinsic) {
+    if (!FLAG_IS_DEFAULT(UseSpinLoopHintIntrinsic))
+      warning("spinLoopHint intrinsic is not available on this CPU");
+    FLAG_SET_DEFAULT(UseSpinLoopHintIntrinsic, false);
+  }
+
   if (FLAG_IS_DEFAULT(ContendedPaddingWidth) &&
     (cache_line_size > ContendedPaddingWidth))
     ContendedPaddingWidth = cache_line_size;
