--- old/src/share/vm/opto/c2compiler.cpp	2016-01-27 16:26:29.000000000 +0300
+++ new/src/share/vm/opto/c2compiler.cpp	2016-01-27 16:26:29.000000000 +0300
@@ -399,6 +399,7 @@
   case vmIntrinsics::_fullFence:
   case vmIntrinsics::_currentThread:
   case vmIntrinsics::_isInterrupted:
+  case vmIntrinsics::_onSpinWait:
 #ifdef TRACE_HAVE_INTRINSICS
   case vmIntrinsics::_classID:
   case vmIntrinsics::_threadID:
