--- old/src/java.base/share/classes/java/lang/Thread.java	2015-10-13 13:38:56.330025006 +0300
+++ new/src/java.base/share/classes/java/lang/Thread.java	2015-10-13 13:38:56.190028313 +0300
@@ -343,6 +343,19 @@
     }
 
     /**
+     * Provide the JVM with a hint that this call is made from within a spinning
+     * loop. The JVM may assume that the speed of executing the loop (e.g. in
+     * terms of number of loop executions per second) is less important than the
+     * reaction time to events that would cause the loop to terminate, or than
+     * potential power savings that may be derived from possible execution
+     * choices. The JVM will not slow down the loop execution to a point where
+     * execution will be delayed indefinitely, but other choices of loop execution
+     * speed are system-specific.
+     */
+    @HotSpotIntrinsicCandidate
+    public static void spinLoopHint() {};
+
+    /**
      * Initializes a Thread with the current AccessControlContext.
      * @see #init(ThreadGroup,Runnable,String,long,AccessControlContext)
      */
