--- old/src/share/vm/opto/library_call.cpp	2015-10-14 14:23:25.000000000 +0300
+++ new/src/share/vm/opto/library_call.cpp	2015-10-14 14:23:25.000000000 +0300
@@ -271,6 +271,7 @@
   bool inline_unsafe_load_store(BasicType type,  LoadStoreKind kind);
   bool inline_unsafe_ordered_store(BasicType type);
   bool inline_unsafe_fence(vmIntrinsics::ID id);
+  bool inline_spinloophint();
   bool inline_fp_conversions(vmIntrinsics::ID id);
   bool inline_number_methods(vmIntrinsics::ID id);
   bool inline_reference_get();
@@ -601,6 +602,8 @@
   case vmIntrinsics::_storeFence:
   case vmIntrinsics::_fullFence:                return inline_unsafe_fence(intrinsic_id());
 
+  case vmIntrinsics::_spinLoopHint:             return inline_spinloophint();
+
   case vmIntrinsics::_currentThread:            return inline_native_currentThread();
   case vmIntrinsics::_isInterrupted:            return inline_native_isInterrupted();
 
@@ -2874,6 +2877,11 @@
   }
 }
 
+bool LibraryCallKit::inline_spinloophint() {
+  insert_mem_bar(Op_SpinLoopHint);
+  return true;
+}
+
 bool LibraryCallKit::klass_needs_init_guard(Node* kls) {
   if (!kls->is_Con()) {
     return true;
@@ -3256,6 +3264,9 @@
     // nothing is an instance of a primitive type
     query_value = gen_instanceof(obj, kls, safe_for_replace);
     break;
+  
+  case vmIntrinsics::_spinLoopHint:
+    break;
 
   case vmIntrinsics::_getModifiers:
     p = basic_plus_adr(kls, in_bytes(Klass::modifier_flags_offset()));
