--- old/src/share/vm/opto/matcher.cpp	2015-10-14 14:23:26.000000000 +0300
+++ new/src/share/vm/opto/matcher.cpp	2015-10-14 14:23:25.000000000 +0300
@@ -939,6 +939,7 @@
     case Op_AryEq:
     case Op_MemBarVolatile:
     case Op_MemBarCPUOrder: // %%% these ideals should have narrower adr_type?
+    case Op_SpinLoopHint:
     case Op_EncodeISOArray:
       nidx = Compile::AliasIdxTop;
       nat = NULL;
