--- old/src/share/vm/opto/memnode.cpp	2015-10-14 14:23:26.000000000 +0300
+++ new/src/share/vm/opto/memnode.cpp	2015-10-14 14:23:26.000000000 +0300
@@ -2891,6 +2891,7 @@
   case Op_MemBarReleaseLock: return new MemBarReleaseLockNode(C, atp, pn);
   case Op_MemBarVolatile:    return new MemBarVolatileNode(C, atp, pn);
   case Op_MemBarCPUOrder:    return new MemBarCPUOrderNode(C, atp, pn);
+  case Op_SpinLoopHint:      return new SpinLoopHintNode(C, atp, pn);
   case Op_Initialize:        return new InitializeNode(C, atp, pn);
   case Op_MemBarStoreStore:  return new MemBarStoreStoreNode(C, atp, pn);
   default: ShouldNotReachHere(); return NULL;
