--- old/src/share/vm/runtime/globals.hpp	2015-11-30 22:53:45.357310590 +0300
+++ new/src/share/vm/runtime/globals.hpp	2015-11-30 22:53:44.799520390 +0300
@@ -858,6 +858,9 @@
   diagnostic(ccstrlist, DisableIntrinsic, "",                               \
          "do not expand intrinsics whose (internal) names appear here")     \
                                                                             \
+  product(bool, UseOnSpinWaitIntrinsic, false,                              \
+          "Use intrinsic for onSpinWait")                                   \
+                                                                            \
   develop(bool, TraceCallFixup, false,                                      \
           "Trace all call fixups")                                          \
                                                                             \
