--- old/src/cpu/x86/vm/x86_64.ad	2015-11-30 22:53:33.333006295 +0300
+++ new/src/cpu/x86/vm/x86_64.ad	2015-11-30 22:53:32.994584072 +0300
@@ -6340,6 +6340,25 @@
   ins_pipe(empty);
 %}
 
+instruct onspinwait()
+%{
+  match(OnSpinWait);
+  ins_cost(200);
+
+  format %{
+    $$template
+    if (os::is_MP()) {
+      $$emit$$"pause\t! membar_onspinwait"
+    } else {
+      $$emit$$"MEMBAR-onspinwait ! (empty encoding)"
+    }
+  %}
+  ins_encode %{
+    __ pause();
+  %}
+  ins_pipe(pipe_slow);
+%}
+
 instruct membar_volatile(rFlagsReg cr) %{
   match(MemBarVolatile);
   effect(KILL cr);
